### PR TITLE
Imageview state control using ViewState class

### LINKED ISSUE #41: About saveState and restoreState
Hi,

   Would saveState return a state class? 
Then if we need, we can use the state class to restore the ImageCropView state.
Because I have a view with the gallery, and I need to restore the Image when I edit the image again...

thank you.

### PR DESCRIPTION
**Pull request by rnitame**
fix #41 
In this PR, it is possible to restore the state of View using ViewState class.

### CODE REVIEW

File: imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java [hunk 1]
```diff
@@ -981,23 +982,25 @@ private int dpToPixel(int dp) {
...
+
+        return new ViewState(getImageViewMatrix(), mAspectRatioWidth, mAspectRatioHeight, suppMatrixValues);
     }
 
-    public void restoreState() {
+    public void restoreState(ViewState viewState) {
         mBitmapChanged = true;
         mRestoreRequest = true;
-        mAspectRatioWidth = savedAspectRatioWidth;
-        mAspectRatioHeight = savedAspectRatioHeight;
```
Comment (helloyako) [inline imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java:994]: @rnitame 
I think `savedAspectRatioWidth`, `savedAspectRatioHeight` and `suppMatrixValues` fields don't need anymore. isn't it?

Comment (rnitame) [reply imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java:994]: `savedAspectRatioWidth` and `savedAspectRatioHeight` fields are unnecessary 🙆 I will delete.
But I could not restore without `suppMatrixValues` field.

Comment (helloyako) [reply imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java:994]: You deleted `aspectRatioWidth` and `aspectRatioHeight` in ViewState. isn't that right?
I mentioned about `savedAspectRatioWidth` and `savedAspectRatioHeight`

Comment (rnitame) [reply imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java:994]: > You deleted aspectRatioWidth and aspectRatioHeight in ViewState. isn't that right?

Right.

> I mentioned about savedAspectRatioWidth and savedAspectRatioHeight

It means this 2 lines? 

https://github.com/naver/android-imagecropview/blob/develop/imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java#L132-L133

Comment (rnitame) [reply imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java:994]: updated https://github.com/naver/android-imagecropview/compare/04a5dfe41e3f88447146936130c5e473f9c2fc30..3f7ad5a525085c45f158d399017d082235e1bc39

**Review by helloyako — APPROVED** [review]

**Comment by helloyako** [comment]
Thank you so much.

Do you need to deploy to maven?

**Comment by rnitame** [comment]
Yes! please 🙏 I will use my product :)

**Comment by helloyako** [comment]
@rnitame 
I deployed 1.2.3-SNAPSHOT to snapshot repository.

Can you test in SNAPSHOT?

### snapshot version from maven
```
repositories {
    maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
}
dependencies {
   implementation 'com.naver.android.helloyako:imagecropview:1.2.3-SNAPSHOT'
}
```

**Comment by rnitame** [comment]
@helloyako Thanks! I confirmed that it was working properly in my product.

**Comment by helloyako** [comment]
It's released.

Thank you!

**Pull request merged**